turbopack/crates/turbopack-core/src/chunk/chunk_group.rs

```diff
@@ -149,36 +163,47 @@ pub async fn make_chunk_group(
             .collect(),
     );
 
-    let mut referenced_output_assets = (*external_output_assets.await?).clone();
-    referenced_output_assets.extend(
-        references_to_output_assets(external_module_references)
-            .await?
-            .await?
-            .iter()
-            .copied(),
-    );
-
-    let rebased_modules = traced_modules
+    let traced_output_assets = traced_modules
         .into_iter()
-        .map(|module| {
-            RebasedAsset::new(
-                *module,
-                module.ident().path().root(),
-                module.ident().path().root(),
-            )
-            .to_resolved()
+        .map(|module| async move {
+            Ok(ResolvedVc::upcast(
+                RebasedAsset::new(
+                    *module,
+                    module.ident().path().root(),
+                    module.ident().path().root(),
+                )
+                .to_resolved()
+                .await?,
+            ))
         })
         .try_join()
         .await?;
 
-    referenced_output_assets.extend(rebased_modules.into_iter().map(ResolvedVc::upcast));
+    let mut chunk_items = all_modules
+        .iter()
+        .map(|(m, async_info)| {
+            (
+                ChunkItemTy::Included.cell(),
+                m.as_chunk_item(chunking_context),
+                *async_info,
+            )
+        })
+        .collect::<Vec<_>>();
+
+    chunk_items.extend(passthrough_modules.into_iter().map(|m| {
+        (
+            ChunkItemTy::Passthrough.cell(),
+            m.as_chunk_item(chunking_context),
+            None,
+        )
+    }));
 
     // Pass chunk items to chunking algorithm
     let mut chunks = make_chunks(
         chunking_context,
-        Vc::cell(chunk_items.into_iter().collect()),
+        Vc::cell(chunk_items),
```

Comment:
I think it would be better to have `chunk_item` and `passthrough_items` as separate lists instead of the type enum.

Reply:
Just tried a few attempts at this -- it duplicates a lot of code and makes things fairly more complex. Maybe we can use that effort to eliminating the need for passthrough items entirely.